Assets/Scripts/Cards/EnemyCards/BeetleCards/Hatchery.cs

```diff
@@ -0,0 +1,40 @@
+using UnityEngine;
+
+
+public class Hatchery : ActionClass
+{
+    public override void ExecuteActionEffect()
+    {
+
+    }
+
+    // Start is called before the first frame update
+    public override void Initialize()
+    {
+        base.Initialize();
+        lowerBound = 2;
+        upperBound = 4;
+
+        Speed = 1;
+
+        description = "Spend +2 resonate to play this card. If this card is unstaggered, spawn 1 random beetle.";
+
+        myName = "Hatchery";
+        CardType = CardType.Defense;
+        Renderer renderer = GetComponent<Renderer>();
+        ogMaterial = renderer.material; // og sprite of card
+        OriginalPosition = transform.position;
+    }
+
+
+    // if origin is a queen beetle, summon a beetle on hit
+    public override void OnHit()
+    {
+        base.OnHit();
+        if (Origin.GetType() == typeof(QueenBeetle))
+        {
+            ((QueenBeetle)Origin).SummonBeetle();
+        }
```

Comment:
changed to cardisunstaggered, i think hard casting here is fine considering the scope of the demo, especially since nothing else has hatchery but lmk if it actually needs to be an event